file.go

```diff
@@ -39,47 +48,52 @@ func (t *TOMLLoader) Load(s interface{}) error {
 }
 
 // JSONLoader satisifies the loader interface. It loads the configuration from
-// the given json file.
+// the given json file or Reader
 type JSONLoader struct {
-	Path string
+	Path   string
+	Reader io.Reader
 }
 
 // Load loads the source into the config defined by struct s
+// Defaults to using the Reader if provided, otherwise tries to read from the
+// file
 func (j *JSONLoader) Load(s interface{}) error {
-	filePath, err := getConfigPath(j.Path)
-	if err != nil {
-		return err
-	}
-
-	file, err := ioutil.ReadFile(filePath)
-	if err != nil {
-		return err
+	var r io.Reader
+	if j.Reader != nil {
+		r = j.Reader
+	} else {
+		file, err := getConfig(j.Path)
+		if err != nil {
+			return err
+		}
+		defer file.Close()
+		r = file
 	}
 
-	return json.Unmarshal(file, s)
+	return json.NewDecoder(r).Decode(s)
 }
 
-func getConfigPath(path string) (string, error) {
+func getConfig(path string) (*os.File, error) {
 	if path == "" {
-		return "", ErrPathNotSet
+		return nil, ErrPathNotSet
 	}
 
 	pwd, err := os.Getwd()
 	if err != nil {
-		return "", err
+		return nil, err
 	}
 
 	configPath := filepath.Join(pwd, path)
 
 	// check if file with combined path is exists(relative path)
 	if _, err := os.Stat(configPath); !os.IsNotExist(err) {
```

Comment:
We should probably avoid calling `os.Stat` at all and just call `os.Open`. Because if it doesn't exists `os.Open` will already return an error.